hnn_core/viz.py

```diff
@@ -0,0 +1,165 @@
+"""Visualization functions."""
+
+# Authors: Mainak Jas <mainak.jas@telecom-paristech.fr>
+#          Sam Neymotin <samnemo@gmail.com>
+
+import numpy as np
+
+
+def plot_dipole(dpl, ax=None, layer='agg', show=True):
+    """Simple layer-specific plot function.
+
+    Parameters
+    ----------
+    dpl : instance of Dipole
+        The Dipole object.
+    ax : instance of matplotlib figure | None
+        The matplotlib axis
+    layer : str
+        The layer to plot. Can be one of
+        'agg', 'L2', and 'L5'
+    show : bool
+        If True, show the figure
+
+    Returns
+    -------
+    fig : instance of plt.fig
+        The matplotlib figure handle.
+    """
+    import matplotlib.pyplot as plt
+
+    if ax is None:
+        fig, ax = plt.subplots(1, 1)
+    if layer in dpl.dpl.keys():
+        ax.plot(dpl.t, dpl.dpl[layer])
+        ax.set_xlabel('Time (ms)')
+        ax.set_title(layer)
+    if show:
+        plt.show()
+    return ax.get_figure()
+
+
+def plot_hist_input(net, ax=None, show=True):
```

Comment:
What do you think of creating a more general function that can plot any spike type (i.e., not just input spikes): `plot_spikes_hist(net, ax=None, spike_types=None, show=True)`? `spike_types` could be entered by the user as a dict of lists s.t. the elements of each list specify which spike types are included in a given category of the histogram. E.g., `spike_types={'proximal inputs': ['evprox_1', 'evprox_2'], 'distal inputs': ['evdist_1']}` would plot the histogram we expect to see for the default.json parameter set. This would establish a much more flexible API that explicitly defines which types (e.g., 'evprox', 'evdist', distal 'common' feed spikes, etc.) are being plotted as discussed in #135.

Reply:
I agree with what you say, do you want me to do this in this PR itself? Or should we leave it for a new contributor as discussed in #135 ? 

I can also do it in another PR myself if I'm getting bored on a Friday night ... I feel this current PR is more of "moving things around"

Reply:
@jasmainak I guess let's save my suggestions for another PR and/or contributor. It should be pretty straight-forward given this PR.

Reply:
Feel free to copy the comment or add a pointer to it in #135 :)